eng/DocsGenerator/Program.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace DocsGenerator;
+
+public class Program
+{
+    public static void Main()
+    {
+        var templatePath = GetRelativeToGitRoot("eng/DocsGenerator/README.template.md");
+
+        var builder = new StringBuilder();
+        using var sr = new StreamReader(templatePath);
+        var line = sr.ReadLine();
+        while (line is not null)
+        {
+            if (line.StartsWith("[!code-"))
+            {
+                var start = line.IndexOf('(');
+                var end = line.IndexOf('#');
+                var filepath = line.Substring(start + 1, end - start - 1);
+                var lines = line.Substring(end + 1, line.Length - end - 1 - ")]".Length).Split('-');
+
+                if (lines.Length != 2)
+                {
+                    throw new ArgumentOutOfRangeException($"Invalid line range in {line}");
+                }
+
+                var startLine = int.Parse(lines[0].Substring("L".Length));
+                var endLine = int.Parse(lines[1].Substring("L".Length));
+                var language = line["[!code-".Length..line.IndexOf("[]")];
+
+                var codeSnippet = GetCodeSnippet(filepath, startLine, endLine, language);
+
+                builder.Append(codeSnippet);
+            }
+            else
+            {
+                builder.AppendLine(line);
+            }
```

Comment:
Personally a fan of early return:
```suggestion
            if (!line.StartsWith("[!code-"))
            {
                builder.AppendLine(line);
                continue;
            }

            var start = line.IndexOf('(');
            var end = line.IndexOf('#');
            var filepath = line.Substring(start + 1, end - start - 1);
            var lines = line.Substring(end + 1, line.Length - end - 1 - ")]".Length).Split('-');

            if (lines.Length != 2)
            {
                throw new ArgumentOutOfRangeException($"Invalid line range in {line}");
            }

            var startLine = int.Parse(lines[0].Substring("L".Length));
            var endLine = int.Parse(lines[1].Substring("L".Length));
            var language = line["[!code-".Length..line.IndexOf("[]")];

            var codeSnippet = GetCodeSnippet(filepath, startLine, endLine, language);

            builder.Append(codeSnippet);
```